cli/src/cmd/cast/call.rs

```diff
@@ -0,0 +1,67 @@
+use crate::opts::{
+    cast::{parse_block_id, parse_name_or_address},
+    EthereumOpts,
+};
+
+use clap::Parser;
+use ethers::types::{BlockId, NameOrAddress};
+use eyre::Result;
+use foundry_config::{
+    figment::{
+        self,
+        value::{Dict, Map, Value},
+        Metadata, Profile,
+    },
+    find_project_root_path, Config,
+};
+
+use serde::Serialize;
+
+#[derive(Debug, Clone, Parser, Serialize)]
+pub struct CallArgs {
+    #[clap(help = "the address you want to query", parse(try_from_str = parse_name_or_address))]
+    #[serde(skip)]
+    pub address: NameOrAddress,
+    #[serde(skip)]
+    pub sig: String,
+    #[serde(skip)]
+    pub args: Vec<String>,
+    #[clap(long, short, help = "the block you want to query, can also be earliest/latest/pending", parse(try_from_str = parse_block_id))]
+    #[serde(skip)]
+    pub block: Option<BlockId>,
+    #[clap(flatten)]
+    #[serde(flatten)]
+    pub eth: EthereumOpts,
+}
+
+impl<'a> From<&'a CallArgs> for Config {
+    fn from(args: &'a CallArgs) -> Self {
+        let config = Config::figment_with_root(find_project_root_path().unwrap())
+            .merge(args)
+            .extract::<Config>()
+            .unwrap_or_else(|err| panic!("{}", err));
+
+        config
+    }
+}
+
+impl figment::Provider for CallArgs {
+    fn metadata(&self) -> Metadata {
+        Metadata::named("Call args provider")
+    }
+
+    fn data(&self) -> Result<Map<Profile, Dict>, figment::Error> {
+        let value = Value::serialize(self)?;
+        let mut dict = value.into_dict().unwrap();
+
+        if let Some(rpc_url) = &self.eth.rpc_url {
+            dict.insert("eth_rpc_url".to_string(), Value::from(rpc_url.to_string()));
+        }
+
+        if let Some(_) = self.eth.from {
+            dict.insert("sender".to_string(), Value::from(dict.get("from").unwrap().clone()));
```

Comment:
shouldn't this be the value of `self.eth.from`?

Reply:
Oh yes, wanted to ask you about this. Yeah I used `self.eth.from` initially, which is a H160, but couldnt figure out how change it into a `Value`. Tried making it a string using to_string() and then passing to `Value` but changing it to string gave weird behaviour, instead of the whole address I got something like "0xD07...123", so I just used value from the dict. ik i'm missing something simple but couldnt find it :(

Reply:
Basically this
```rust
 if let Some(from) = self.eth.from {
            dict.insert("sender".to_string(), Value::from(from.to_string())); 
```
This worked but gave the wrong value for sender like "0xabc...124". Couldn't find docs for this too.

Reply:
yeah this is a bit weird, you need to invoke the debug implementation with `format!("{:?}", from)`

Reply:
ty that works,made the changes :)